examples/complete/main.tf

```diff
@@ -437,3 +437,30 @@ module "prefix_list" {
     },
   ]
 }
+
+#################################
+# Security group using prefix list
+#################################
+resource "aws_ec2_managed_prefix_list" "prefix_list_sg_example" {
+  address_family = "IPv4"
+  max_entries    = 1
+  name           = "prefix-list-sg-example"
+
+  entry {
+    cidr        = module.vpc.vpc_cidr_block
+    description = "VPC CIDR"
+  }
+}
+
+module "prefix_list_sg" {
+  source = "../../"
+
+  name        = "prefix-list-sg"
+  description = "Security group using prefix list"
+  vpc_id      = data.aws_vpc.default.id
+
+  use_name_prefix = false
+
+  ingress_prefix_list_ids = [aws_ec2_managed_prefix_list.prefix_list_sg_example.id]
```

Comment:
Please use `ingress_with_prefix_list_ids` instead in the example to show the proposed feature for real.